### PR TITLE
Add Emscripten barebone build implementation

### PR DESCRIPTION
**Pull request by emaxx-google**
This adds the first version of Emscripten-specific Makefile definitions,
allowing to compile the C/C++ code under WebAssembly. This new
functionality can be enabled by setting the TOOLCHAIN environment
variable to "emscripten" (the variable still defaults to "pnacl" to use
Google Native Client, as before).

Note that this is only a barebone implementation, since it doesn't
support several important features, namely:
* Auto-detecting related .h files and triggering recompilation of .cc
  files on their changes;
* Installation of headers into shared location;
* Building unit tests;
* etc.

These will be addressed in follow-ups, as tracked by #177.

Another note - Right now only very few files in the project can be
compiled using this new Emscripten toolchain, since many files are using
Native Client specific definitions (like the pp::Var class). Their
migration is a separate ongoing effort.

### CODE REVIEW

File: common/make/internal/executable_building_emscripten.mk [hunk 1]
```diff
@@ -0,0 +1,195 @@
+# Copyright 2016 Google Inc. All Rights Reserved.
```
Comment (MahmoudAGawad) [inline common/make/internal/executable_building_emscripten.mk:1]: nit: 2020

File: common/make/internal/executable_building_emscripten.mk [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+
+# Flags passed to the Emscripten compiler/linker tools.
+#
+# Explanation:
+# bind: Enables "Embind" (the technology used by our C++ wrappers for talking
+#   with JavaScript).
+# pthread: Enables Pthreads support (for C/C++ multi-threading, etc.).
+# MODULARIZE: Puts Emscripten module JavaScript loading code into a factory
+#   function, in order to control its loading from other JS code and to avoid
+#   avoid name conflicts with unrelated code.
```
Comment (MahmoudAGawad) [inline common/make/internal/executable_building_emscripten.mk:63]: nit: remove repeated "avoid"

File: common/make/internal/executable_building_emscripten.mk [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+ifeq ($(CONFIG),Release)
+
+# Add flags specific to release builds.
+#
+# Explanation:
+# O3: Enable advanced optimizations.
+EMSCRIPTEN_FLAGS += \
+	-O3 \
+
+else ifeq ($(CONFIG),Debug)
```
Comment (MahmoudAGawad) [inline common/make/internal/executable_building_emscripten.mk:84]: not related to this line but I find that using the `EMCC_DEBUG` env variable can be helpful while in Debug mode: https://emscripten.org/docs/porting/Debugging.html#debug-mode-emcc-debug

**Review by MahmoudAGawad — APPROVED** [review]
LGTM % nits

**Review by Fabian-Sommer — APPROVED** [review]
LGTM % nit

**Comment by emaxx-google** [comment]
Thanks!

**Pull request merged**